alf/algorithms/sac_algorithm.py

```diff
@@ -224,63 +219,54 @@ def _actor_train_step(self, exp: Experience, state: SacActorState,
         if self._is_continuous:
             critic_input = (exp.observation, action)
 
-            with tf.GradientTape(watch_accessed_variables=False) as tape:
-                tape.watch(action)
-                critic1, critic1_state = self._critic_network1(
-                    critic_input,
-                    step_type=exp.step_type,
-                    network_state=state.critic1)
-
-                critic2, critic2_state = self._critic_network2(
-                    critic_input,
-                    step_type=exp.step_type,
-                    network_state=state.critic2)
+            critic1, critic1_state = self._critic_network1(
+                critic_input, state=state.critic1)
+            critic2, critic2_state = self._critic_network2(
+                critic_input, state=state.critic2)
 
-                target_q_value = tf.minimum(critic1, critic2)
+            target_q_value = torch.min(critic1, critic2)
 
-            dqda = tape.gradient(target_q_value, action)
+            dqda = torch.autograd.grad(target_q_value, action,
```

Comment:
action might be nested.

Reply:
Good point. Changed to properly handle nested action.